It's related to your change, but cut out of the log:
```
jax/_src/numpy/lax_numpy.py:3172: error: Argument 1 to "len" has incompatible type "int | Sequence[int] | Sequence[Sequence[int]]"; expected "Sized"  [arg-type]
jax/_src/numpy/lax_numpy.py:3173: error: "int" object is not iterable  [misc]
jax/_src/numpy/lax_numpy.py:3178: error: Argument 1 to "len" has incompatible type "Any | int | Sequence[int]"; expected "Sized"  [arg-type]
jax/_src/numpy/lax_numpy.py:3181: error: Item "int" of "Any | int | Sequence[int]" has no attribute "__iter__" (not iterable)  [union-attr]
jax/_src/numpy/lax_numpy.py:3182: error: Item "int" of "Any | int | Sequence[int]" has no attribute "__iter__" (not iterable)  [union-attr]
jax/experimental/shard_map.py:1019: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
Found 5 errors in 1 file (checked 1 source file)
```

--- a/jax/_src/numpy/lax_numpy.py
+++ b/jax/_src/numpy/lax_numpy.py
@@ -3156,7 +3156,9 @@ def vdot(a, b, *, precision=None):
 
 
 @util._wraps(np.tensordot, lax_description=_PRECISION_DOC)
-def tensordot(a, b, axes=2, *, precision=None):
+def tensordot(a: ArrayLike, b: ArrayLike,
+              axes: Union[int, Sequence[int], Sequence[Sequence[int]]] = 2,
+              *, precision=None) -> Array:
   util.check_arraylike("tensordot", a, b)
   a_ndim = ndim(a)
   b_ndim = ndim(b)
@@ -3167,12 +3169,12 @@ def tensordot(a, b, axes=2, *, precision=None):
       msg = "Number of tensordot axes (axes {}) exceeds input ranks ({} and {})"
       raise TypeError(msg.format(axes, a.shape, b.shape))
     contracting_dims = tuple(range(a_ndim - axes, a_ndim)), tuple(range(axes))
-  elif type(axes) in (list, tuple) and len(axes) == 2:
+  elif isinstance(axes, (tuple, list)) and len(axes) == 2:
     ax1, ax2 = axes
     if type(ax1) == type(ax2) == int:
       contracting_dims = ((_canonicalize_axis(ax1, a_ndim),),
                           (_canonicalize_axis(ax2, b_ndim),))
-    elif type(ax1) in (list, tuple) and type(ax2) in (list, tuple):
+    elif isinstance(ax1, (tuple, list)) and isinstance(ax2, (tuple, list)):
       if len(ax1) != len(ax2):
         msg = "tensordot requires axes lists to have equal length, got {} and {}."
         raise TypeError(msg.format(ax1, ax2))

